src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,423 @@
+    });
+  }
+
+  getFirstTabAndField() {
+    const { packageDetails } = this.props;
+    const schema = packageDetails.getConfig();
+
+    const [activeTab] = Object.keys(schema.properties);
+    const [focusField] = Object.keys(schema.properties[activeTab].properties);
+
+    return {
+      activeTab,
+      focusField
+    };
+  }
+
+  handleJSONToggle() {
+    this.setState({ jsonEditorActive: !this.state.jsonEditorActive });
+  }
+
+  handleCloseConfirmModal() {
+    this.setState({ isConfirmOpen: false });
+  }
+
+  handleGoBack() {
+    const { reviewActive, hasChangesApplied } = this.state;
+
+    if (reviewActive && hasChangesApplied) {
+      this.setState({ reviewActive: false });
+
+      return;
+    }
+
+    if (hasChangesApplied) {
+      this.setState({ isConfirmOpen: true });
+
+      return;
+    }
+
+    this.handleConfirmGoBack();
+  }
+
+  handleServiceReview() {
+    const { handleRun } = this.props;
+    const { reviewActive } = this.state;
+
+    if (reviewActive) {
+      handleRun();
+
+      return;
+    }
+
+    this.setState({ reviewActive: true });
+  }
+
+  handleConfirmGoBack() {
+    const { handleGoBack } = this.props;
+
+    this.setState({ isConfirmOpen: false, isOpen: false }, () => {
+      // Once state is set, start a timer for the length of the animation and
+      // navigate away once the animation is over.
+      setTimeout(handleGoBack, 300);
+    });
+  }
+
+  getSecondaryActions() {
+    const { reviewActive, hasChangesApplied } = this.state;
+
+    return [
+      {
+        className: "button-stroke",
+        clickHandler: this.handleGoBack,
+        label: reviewActive && hasChangesApplied ? " Back" : "Cancel"
+      }
+    ];
+  }
+
+  getPrimaryActions() {
+    const { jsonEditorActive, reviewActive } = this.state;
+    const { formErrors } = this.props;
+
+    const hasFormErrors = Object.keys(formErrors).reduce(
+      (sum, tab) => sum + formErrors[tab],
+      0
+    );
+
+    const actions = [
+      {
+        className: "button-primary flush-vertical",
+        clickHandler: this.handleServiceReview,
+        label: reviewActive ? "Run Service" : "Review & Run",
+        disabled: hasFormErrors
+      }
+    ];
+
+    if (!reviewActive) {
+      actions.unshift({
+        node: (
+          <ToggleButton
+            className="flush"
+            checkboxClassName="toggle-button toggle-button-align-left"
+            checked={jsonEditorActive}
+            onChange={this.handleJSONToggle}
+            key="json-editor"
+          >
+            JSON Editor
+          </ToggleButton>
+        )
+      });
+    }
+
+    return actions;
+  }
+
+  getReviewScreen() {
+    const {
+      packageDetails,
+      deployErrors,
+      isInitialDeploy,
+      formData
+    } = this.props;
+
+    const fileName = "config.json";
+    const configString = JSON.stringify(formData, null, 2);
+
+    const renderKeys = {};
+    this.getHashMapRenderKeys(formData, renderKeys);
```

Comment:
Please update this method so it returns the `renderKeys` and doesn't mutate a previously set const.